### DISCUSSION TITLE
Multi variable select

### DISCUSSION THREAD

**Discussion by rkalayn3** [grafana/grafana-infinity-datasource#655] [Q&A]
Greetings @yesoreyeram , 

I have defined a variable called 'Severity' with values - Critical, High, Medium, Low, Informational. I was able to reference this variable in the dashboard metric. The visualization is loading data correctly if I select a single value. However, It is not showing any data if I select multi value or select 'ALL'. 

<img width="460" alt="image" src="https://github.com/yesoreyeram/grafana-infinity-datasource/assets/142121482/813721cc-c2d5-4d47-9738-b6bf218bd218">

This is how I am referencing the variable:

<img width="725" alt="image" src="https://github.com/yesoreyeram/grafana-infinity-datasource/assets/142121482/34741bd1-0214-46ef-80c0-cddb6630e3f7">

<img width="698" alt="image" src="https://github.com/yesoreyeram/grafana-infinity-datasource/assets/142121482/f081df70-73fd-43c4-97fc-f5a3be903d0b">

**Comment by yesoreyeram** [comment]
Hiya.. that should be possible. Have you tried IN command in your filter? Backend documentation have examples and screenshots as well.

**Reply by rkalayn3** [reply]
@yesoreyeram I tried and it is giving me an error

<img width="675" alt="image" src="https://github.com/yesoreyeram/grafana-infinity-datasource/assets/142121482/3a7513bf-beab-4649-9680-ff2d0bf2ec42">

**Reply by yesoreyeram** [reply]
You are almost there... Just you needed to use correct variable format. And something like  `Sev IN (${sev:singleq})` format. https://grafana.com/docs/grafana/latest/dashboards/variables/variable-syntax/#singlequote

Note: sorry for the typos.. typing from mobile..

**Reply by yesoreyeram** [reply]
`SeverityNum IN (${Severity:singlequote})`

**Reply by rkalayn3** [reply]
@yesoreyeram thanks. I tried this and I am still getting the same error as above

<img width="692" alt="image" src="https://github.com/yesoreyeram/grafana-infinity-datasource/assets/142121482/c6abc5b8-87a1-405e-acfd-e3ba7d77fc04">

**Reply by yesoreyeram** [reply]
Can you help me with the result JSON sample data?

**Reply by rkalayn3** [reply]
@yesoreyeram, here is a sample. I have several fields but I am just providing a sample that has SeverityNum as one of the field. It is of type string with values 1-Critcal, 2-High, 3-Medium and 4-low. The Total Open count in the above image is count of all "AppName"

{
        "AppName": "acdety",
        "LastModifiedDate": "2023-08-11T18:08:28.120Z",
        "ResolveETA": "TBD",
        "ResolveDate": "TBD",
        "Solution": "",
        "CloudType": "Prd",
        "SeverityNum": "1-Critical",
        "ClosedDate": "TBD",
        "IssueID": "xycjdlsd9434fm"
    }

**Reply by rkalayn3** [reply]
It worked for multi select but not for single select. Can you help me here? @yesoreyeram 
I used below syntax

Source in (${Source:singlequote})

The variable I am referring is ‘Source’. It has 3 values – All, PBot and SNOW. Works if I select PBot +SNOW but does not work if I select just “PBot” or just “SNOW”.

**Reply by jacob623** [reply]
Was there ever a resolution to this @rkalayn3 @yesoreyeram? I am experiencing the same. When I use "in", it works when multiple values are selected, however does not with just a single selected value.

**Reply by yesoreyeram** [reply]
Sorry @rkalayn3 for the delay. Here is the example of filters. I am using https://restcountries.com/v3.1/all

## single value filter

<img width="1441" alt="image" src="https://github.com/yesoreyeram/grafana-infinity-datasource/assets/153843/18a43f8b-aabe-400e-b72d-20a464a94ba0">

## multi value filter

for the multi value, I am using `region IN (${region_multi:singlequote})` syntax. Related documents [here](https://sriramajeyam.com/grafana-infinity-datasource/docs/backend#filter) and [here](https://grafana.com/docs/grafana/v8.5/variables/advanced-variable-format-options/)

<img width="1388" alt="image" src="https://github.com/yesoreyeram/grafana-infinity-datasource/assets/153843/13e32163-160b-4c1d-9be4-7fbbc7880fa7">

**Reply by rkalayn3** [reply]
@yesoreyeram Does your query work if you select single value using region_multi filter? example: Europe?. I do not see anything wrong with my syntax and it is failing if I select single value. Please advise.

**Reply by jpadmin** [reply]
@rkalayn3 is right! There should be a bug when single value is opted, but as a workaround, please use `region IN (${region_multi:singlequote}, '')`. Basically the idea is you can replace blank with any non existent value.
